packages/libs/error-stack/tests/test_compatibility.rs

```diff
@@ -121,12 +161,17 @@ fn anyhow_output() {
     assert_eq!(anyhow_display_extended, context_display_extended);
 }
 
-#[cfg(feature = "eyre")]
+#[cfg(all(feature = "eyre"))]
 fn install_eyre_hook() {
-    static ONCE: std::sync::Once = std::sync::Once::new();
-    ONCE.call_once(|| {
-        eyre::set_hook(Box::new(eyre::DefaultHandler::default_with)).expect("Could not set hook");
-    })
+    static ONCE: core::sync::atomic::AtomicBool = core::sync::atomic::AtomicBool::new(false);
+
+    if ONCE
+        .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+        .is_ok()
+    {
+        eyre::set_hook(alloc::boxed::Box::new(eyre::DefaultHandler::default_with))
+            .expect("Could not set hook");
+    }
```

Comment:
What was wrong with using `std::sync::Once`?

Reply:
Can we revert it then back to `Once`? In tests, `std` is always available, even if the feature is not specified. However, gated features in the library won't be available without the `std` feature. You can see `std` as an implicit dev-dependency.

Reply:
Yes, already did in one of my upcoming commits.